### PR TITLE
8274328: C2: Redundant CFG edges fixup in block ordering

### PR DESCRIPTION
**Pull request by kelthuzadx**
I think Trace::fixup_blocks is redundant because PhaseCFG::fixup_flow will nevertheless fix up the CFG flow(i.e. flip successor blocks of IfNode) right after PhaseBlockLayout pass, we can remove this step when doing PhaseBlockLayout pass.(Testing: jtreg/compiler/c2, presubmit test)

https://github.com/openjdk/jdk/blob/5ec1cdcaf39229a7d2457313600b0dc2bf8c6453/src/hotspot/share/opto/compile.cpp#L2765

https://github.com/openjdk/jdk/blob/5ec1cdcaf39229a7d2457313600b0dc2bf8c6453/src/hotspot/share/opto/block.cpp#L1679

https://github.com/openjdk/jdk/blob/5ec1cdcaf39229a7d2457313600b0dc2bf8c6453/src/hotspot/share/opto/block.cpp#L908-L916

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274328](https://bugs.openjdk.java.net/browse/JDK-8274328): C2: Redundant CFG edges fixup in block ordering

### Reviewers
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5705/head:pull/5705` \
`$ git checkout pull/5705`

Update a local copy of the PR: \
`$ git checkout pull/5705` \
`$ git pull https://git.openjdk.java.net/jdk pull/5705/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5705`

View PR using the GUI difftool: \
`$ git pr show -t 5705`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5705.diff">https://git.openjdk.java.net/jdk/pull/5705.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/block.cpp [hunk 1]
```diff
@@ -914,7 +914,6 @@ void PhaseCFG::fixup_flow() {
         ProjNode* tmp = proj0;
         proj0 = proj1;
         proj1 = tmp;
```
[the remark this answers is not in the document]
Comment (kelthuzadx) [reply src/hotspot/share/opto/block.cpp]: Thanks Tobias for review. I have replaced above code with `swap(pointer ref,pointer ref)`

**Review by TobiHartmann — APPROVED** [review]
Thanks, looks good.

**Comment by vnkozlov** [comment]
> That looks good to me but I'm not an expert in that code. I submitted some testing and it all passed.

@TobiHartmann please add links to testing in RFE.

**Review by vnkozlov — APPROVED** [review]
Looks good.

**Comment by kelthuzadx** [comment]
Thanks @TobiHartmann and @vnkozlov for reviews!

/integrate

**Comment by TobiHartmann** [comment]
> @TobiHartmann please add links to testing in RFE.

Done.

**Pull request closed without merging**